### ISSUE TITLE
Weird bug where all suits a on only ONE page

### ISSUE THREAD

**Issue by ATRK23** [VerityIncorporated/TooManySuits#22]
<img width="1907" alt="Capture d’écran 2024-09-20 214331" src="https://github.com/user-attachments/assets/66e1342e-1e58-4c30-8a52-fbb0be78350e">

**Comment by ATRK23** [comment]
the pages dont seem to appear anymore

**Comment by notpeelz** [comment]
Can you share your profile code and the log file?
It's hard to say what went wrong without that information.

**Comment by SpiderOver** [comment]
I’m having a similar issue. For me, the pages and the rack works normally. However, when I’m the host, other players can’t see the pages and the rack looks like the one in the OP, extending far away from the ship. When other players are the host, the same thing happens to me, the mod works for them, but not for me.

**Comment by notpeelz** [comment]
Code? Logs?

**Comment by SpiderOver** [comment]
> Code? Logs?
I tried to create a second profile and run only TooManyCosmetics and it worked, so its a compatibility problem with some other mods. Here is the log (I guess that it is this one)
[LogOutput.log](https://github.com/user-attachments/files/17085471/LogOutput.log)

**Comment by notpeelz** [comment]
> However, when I’m the host, other players can’t see the pages and the rack looks like the one in the OP

I did my testing with only 2 players and I didn't have any issues. How many players did you have when they started encountering the issue? Was it a mid-game join (aka "late-join")? Are you using mods to increase the lobby capacity beyond the vanilla limit?
Ideally I would need the log file from the players that encountered the issue, not the host.

**Comment by SpiderOver** [comment]
> > However, when I’m the host, other players can’t see the pages and the rack looks like the one in the OP
> 
> I did my testing with only 2 players and I didn't have any issues. How many players did you have when they started encountering the issue? Was it a mid-game join (aka "late-join")? Are you using mods to increase the lobby capacity beyond the vanilla limit? Ideally I would need the log file from the players that encountered the issue, not the host.

I was also playing with one more person, and it wasn’t a late-join; they joined right at the beginning. And yes, I’m using mods to increase the limit (MC). I’ll try removing it. At the moment, I don’t have anyone to test with, but I’ll create a third profile and run two LCs simultaneously. If removing the MC doesn’t fix the issue, I’ll send the second player log.

**Comment by SpiderOver** [comment]
> Ideally I would need the log file from the players that encountered the issue, not the host.

Yeah, removing MC didn't work. I thought about it more and remembered that it’s also present in TooManyCosmetics, but I managed to get the log from the second player tho.
[LogOutput.log](https://github.com/user-attachments/files/17085530/LogOutput.log)

**Comment by notpeelz** [comment]
I'm not seeing anything obvious in that log file.
We can test it together if you want. I messaged you on discord.

Oh also, if you're running two instances from the same profile, I think they'll end up fighting over control of the log file.
I think you would have to create two separate profiles, so each instance gets its own log file (preferably using [Gale](https://github.com/Kesomannen/gale) with the "Direct" launch mode).

**Comment by notpeelz** [comment]
@SpiderOver and I worked it out in Discord DMs. It turns out it was caused by [MinecraftCaveSounds](https://thunderstore.io/c/lethal-company/p/Bingle/MinecraftCaveSounds/) crashing in its `StartOfRound.Start` patch. This can happen with any mod that patches the same method.

For any future readers: check your logs and pay close attention to errors (even if it seems completely unrelated to TooManySuits).
